dsls/sql/src/main/java/org/apache/beam/dsls/sql/transform/BeamBuiltinAggregations.java

```diff
@@ -383,6 +384,144 @@ public T result(KV<BigDecimal, Long> accumulator) {
     }
   }
 
+  static class VarAgg implements Serializable {
+    long count; // number of elements
+    double sum; // sum of elements
+
+    public VarAgg(long count, double sum) {
+      this.count = count;
+      this.sum = sum;
+    }
+  }
+
+  public static final class Var<T> extends BeamSqlUdaf<T, KV<BigDecimal, VarAgg>, T> {
+    public static Var create(SqlTypeName fieldType, boolean isSamp) {
+      switch (fieldType) {
+        case INTEGER:
+          return new Var<Integer>(fieldType, isSamp);
+        case SMALLINT:
+          return new Var<Short>(fieldType, isSamp);
+        case TINYINT:
+          return new Var<Byte>(fieldType, isSamp);
+        case BIGINT:
+          return new Var<Long>(fieldType, isSamp);
+        case FLOAT:
+          return new Var<Float>(fieldType, isSamp);
+        case DOUBLE:
+          return new Var<Double>(fieldType, isSamp);
+        case DECIMAL:
+          return new Var<BigDecimal>(fieldType, isSamp);
+        default:
+          throw new UnsupportedOperationException(
+                  String.format("[%s] is not support in Var", fieldType));
+      }
+    }
+
+    private SqlTypeName fieldType;
+    private boolean isSamp;
+
+    private Var(SqlTypeName fieldType, boolean isSamp) {
+      this.fieldType = fieldType;
+      this.isSamp = isSamp;
+    }
+
+    @Override
+    public KV<BigDecimal, VarAgg> init() {
+      VarAgg varagg = new VarAgg(0L, 0);
+      return KV.of(new BigDecimal(0), varagg);
+    }
+
+    @Override
+    public KV<BigDecimal, VarAgg> add(KV<BigDecimal, VarAgg> accumulator, T input) {
+      double v;
+      if (input == null) {
+        return accumulator;
+      } else {
+        v = new BigDecimal(input.toString()).doubleValue();
+        accumulator.getValue().count++;
+        accumulator.getValue().sum += v;
+
+        double variance;
+        if (accumulator.getValue().count > 1) {
+          double t = accumulator.getValue().count * v - accumulator.getValue().sum;
+          variance = (t*t)/(accumulator.getValue().count * (accumulator.getValue().count - 1));
+        } else {
+          variance = 0;
+        }
+        return KV.of(accumulator.getKey().add(new BigDecimal(variance)), accumulator.getValue());
+      }
+    }
+
+    @Override
+    public KV<BigDecimal, VarAgg> merge(Iterable<KV<BigDecimal, VarAgg>> accumulators) {
```

Comment:
Should we implement unit tests or integration tests for all these builtin aggregation functions?